pennylane/ops/op_math/adjoint_class.py

```diff
@@ -224,3 +232,20 @@ def diagonalizing_gates(self):
 
     def adjoint(self):
         return self.base.queue()
+
+    @property
+    def arithmetic_depth(self) -> int:
+        return 1 + self.base.arithmetic_depth
+
+    def simplify(self, depth=-1):
+        # TODO: Should we add a self.do_queue attribute and instantiate the simplified class
```

Comment:
What we might want to do is specify `do_queue=False` whenever an operation is constructed within `simplify`.  Then, we manually queue the result in `qml.simplify` via `op.queue()`.

We could also just ignore `do_queue` in the `op.simplify` method and use `qml.tape.stop_recording()` in the `qml.simplify` method.

Reply:
Change added in PR https://github.com/PennyLaneAI/pennylane/pull/2854